core/header.cpp

```diff
@@ -169,6 +169,15 @@ namespace MR
       if (App::project_version)
         H.keyval()["project_version"] = App::project_version;
 
+      std::string cmd = App::argv[0];
+      for (int n = 1; n < App::argc; ++n)
+        cmd += std::string(" \"") + App::argv[n] + "\"";
```

Comment:
Reminds me of [here](https://github.com/MRtrix3/mrtrix3/blob/master/lib/mrtrix3/path.py#L35-L37), where I wrap paths in quotation marks only when necessary for `shlex.split()`. Admittedly that maybe needs to check for more than just whitespace; but that's what triggered the implementation. Why not simply use a similar heuristic here, i.e. only wrap a string / path in quotation marks if it contains special characters (which will be the vast minority of the case)?